### PR TITLE
Update SM vs Cloud features compatibility

### PR DESCRIPTION
**Pull request by micheleRP**
## Description

Resolves https://redpandadata.atlassian.net/browse/DOC-754
Review deadline: Nov 27

## Page previews
[Redpanda Cloud vs Self-Managed feature compatibility](https://deploy-preview-122--rp-cloud.netlify.app/redpanda-cloud/get-started/cloud-overview/#redpanda-cloud-vs-self-managed-feature-compatibility)

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Review by Feediver1 — APPROVED** [review]
love the intro. lgtm

**Review by Feediver1 — APPROVED** [review]
Lgtm

File: modules/get-started/pages/cloud-overview.adoc [hunk 1]
```diff
@@ -245,11 +250,11 @@ Redpanda Cloud deployments do not support the following functionality available
 ** `rpk debug` 
 ** `rpk iotune`
 ** `rpk redpanda`
-** `rpk topic describe-storage` (all other `rpk topic` commands are supported on both Redpanda Cloud and self-managed)
-** `rpk transform`  (currently in beta for Redpanda Cloud)
+** `rpk topic describe-storage` (all other `rpk topic` commands are supported on both Redpanda Cloud and Self Managed)
```
Comment (Deflaimun) [inline modules/get-started/pages/cloud-overview.adoc:253]:
```suggestion
** `rpk topic describe-storage` (all other `rpk topic` commands are supported on both Redpanda Cloud and Self-Managed)
```

Does SM always have to have hyphen?

**Review by Deflaimun — APPROVED** [review]
approved. See comment on SM before merging

File: modules/get-started/pages/cloud-overview.adoc (hunk 1 above)
Comment (micheleRP) [reply modules/get-started/pages/cloud-overview.adoc]: We hyphenate it when it's used as an adjective, like "Self-Managed deployments" or "Self-Managed users". But we do not hyphenate it when it's used as a noun, like "Redpanda Self Managed".

**Comment by Feediver1** [comment]
FYI, may want to add that `rpk debug remote-bundle` commands are not supported in Cloud, which is fully managed. (confirmed with @deniscoady ).

File: modules/get-started/pages/cloud-overview.adoc [hunk 2]
```diff
@@ -220,17 +220,22 @@ NOTE: To create a Redpanda cluster in your virtual private cloud (VPC), follow t
...
-- Data transforms (currently in beta for Redpanda Cloud)
-- Remote Read Replicas (currently in beta for Redpanda Cloud)
-- Kafka API OIDC authentication (however, Redpanda Cloud does support xref:security:cloud-authentication.adoc#single-sign-on[user authentication to an organization with SSO])
-- Kerberos authentication
-- Redpanda debug bundles
-- Redpanda Console topic documentation
-- Setting `auto_create_topics_enabled=true`
-- Admin API
+- Integration with Apache Iceberg. For private beta access on BYOC, contact https://support.redpanda.com/hc/en-us/requests/new[Redpanda support^].
+- Data transforms. This is in beta for Redpanda Cloud.
```
Comment (mattschumpert) [inline modules/get-started/pages/cloud-overview.adoc:228]: We should have the same private beta CTA here

File: modules/get-started/pages/cloud-overview.adoc [hunk 3]
```diff
@@ -220,17 +220,22 @@ NOTE: To create a Redpanda cluster in your virtual private cloud (VPC), follow t
...
+- Remote Read Replicas. This is in beta for Redpanda Cloud.
```
Comment (mattschumpert) [inline modules/get-started/pages/cloud-overview.adoc:229]: same here as well

File: modules/get-started/pages/cloud-overview.adoc [hunk 4]
```diff
@@ -220,17 +220,22 @@ NOTE: To create a Redpanda cluster in your virtual private cloud (VPC), follow t
...
+- Kafka API OIDC authentication. However, Redpanda Cloud does support xref:security:cloud-authentication.adoc#single-sign-on[user authentication to an organization with SSO].
```
Comment (mattschumpert) [inline modules/get-started/pages/cloud-overview.adoc:230]: I think we want to say 'Does support SSO for the Redpanda Cloud UI' @deniscoady ?

**Review by mattschumpert — APPROVED** [review]

File: modules/get-started/pages/cloud-overview.adoc (hunk 2 above)
Comment (micheleRP) [reply modules/get-started/pages/cloud-overview.adoc:228]: @mattschumpert is data transforms in private beta for just BYOC or for all of Redpanda Cloud?

File: modules/get-started/pages/cloud-overview.adoc (hunk 3 above)
Comment (micheleRP) [reply modules/get-started/pages/cloud-overview.adoc:229]: @mattschumpert A difference with RRR is that we have had doc for RRR in Cloud: We've never had any doc for transforms in Cloud. Can you please confirm that for RRR, in addition to including the beta tags on the doc, we should add the admonition that they need to contact support to enable it?

File: modules/get-started/pages/cloud-overview.adoc (hunk 4 above)
Comment (deniscoady) [reply modules/get-started/pages/cloud-overview.adoc:230]: Agreed. Console/Cloud UI supports SSO.

**Comment by micheleRP** [comment]
merging now for 24.3 updates. @mattschumpert we'll make any additional updates in a separate PR.

**Pull request merged**